waspc/packages/wasp-config/__tests__/cli.unit.test.ts

```diff
@@ -0,0 +1,110 @@
+import { describe, expect, test } from 'vitest'
+import { parseProcessArgsOrThrow } from '../src/cli'
+
+describe('parseProcessArgsOrThrow', () => {
+  test('should parse arguments correctly', () => {
+    testParseProcessArgsOrThrow([
+      'node',
+      'run.js',
+      'main.wasp.js',
+      'output.json',
+      JSON.stringify(['entity1']),
+    ])
+  })
+
+  test('should parse 0 entities correctly', () => {
+    testParseProcessArgsOrThrow([
+      'node',
+      'run.js',
+      'main.wasp.js',
+      'output.json',
+      '[]',
+    ])
+  })
+
+  test('should throw an error if less than 5 arguments', () => {
+    testParseProcessArgsOrThrow(
+      ['node', 'run.js', 'main.wasp.js', 'output.json'],
+      { shouldError: true }
+    )
+  })
+
+  test('should throw an error if more than 5 arguments', () => {
+    testParseProcessArgsOrThrow(
+      ['node', 'run.js', 'main.wasp.js', 'output.json', '[]', 'extraArg'],
+      { shouldError: true }
+    )
+  })
+
+  test('should throw an error if waspTsSpecPath is not a string', () => {
+    testParseProcessArgsOrThrow(
+      [
+        'node',
+        'run.js',
+        undefined,
+        'output.json',
+        JSON.stringify(['entity1']),
+      ] as string[],
+      { shouldError: true }
+    )
+  })
+
+  test('should throw an error if outputFilePath is not a string', () => {
+    testParseProcessArgsOrThrow(
+      [
+        'node',
+        'run.js',
+        'main.wasp.js',
+        undefined,
+        JSON.stringify(['entity1']),
+      ] as string[],
+      { shouldError: true }
+    )
+  })
+
+  test('should throw an error if any entityNames is not a string', () => {
+    testParseProcessArgsOrThrow(
+      ['node', 'run.js', 'main.wasp.js', 'output.json', undefined] as string[],
+      { shouldError: true }
+    )
+  })
+
+  test('should throw an error if the entity names JSON is not an array', () => {
+    testParseProcessArgsOrThrow(
+      [
+        'node',
+        'run.js',
+        'main.wasp.js',
+        'output.json',
+        JSON.stringify({ entity1: 'entity1' }),
+      ],
+      { shouldError: true }
+    )
+  })
+
+  function testParseProcessArgsOrThrow(
+    args: string[],
+    options:
```

Comment:
I wonder if it would be just easier if you didn't wrap it in this extra `options`.